src/test/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4KitTest.kt

```diff
@@ -121,6 +144,422 @@ class GoogleAnalyticsFirebaseGA4KitTest {
         TestCase.assertEquals(1, firebaseSdk.loggedEvents.size)
     }
 
+
+    @Test
+    fun onConsentStateUpdatedTest() {
+        val map = HashMap<String, String>()
+        map["defaultAdStorageConsentSDK"] = "Granted"
+        map["defaultAnalyticsStorageConsentSDK"] = "Granted"
+        map["consentMappingSDK"] =
+            "[{\\\"jsmap\\\":null,\\\"map\\\":\\\"Performance\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_user_data\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"Marketing\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_personalization\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"testconsent\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_storage\\\"}]"
+        map["defaultAdUserDataConsentSDK"] = "Denied"
+        map["defaultAdPersonalizationConsentSDK"] = "Denied"
+
+        kitInstance.configuration =
+            KitConfiguration.createKitConfiguration(JSONObject().put("as", map.toMutableMap()))
+
+
+        val locationCollectionConsent = GDPRConsent.builder(false)
+            .document("Test consent")
+            .location("17 Cherry Tree Lane")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+        val state = ConsentState.builder()
+            .addGDPRConsentState("Marketing", locationCollectionConsent)
+            .build()
+        filteredMParticleUser = FilteredMParticleUser.getInstance(user, kitInstance)
+
+        kitInstance.onConsentStateUpdated(state, state, filteredMParticleUser)
+
+        val expectedConsentValue =
+            firebaseSdk.getConsentState().getKeyByValue("AD_PERSONALIZATION").toString()
+        TestCase.assertEquals("DENIED", expectedConsentValue)
```

Comment:
Pardon my ignorance since I'm not as fluent in Kotlin, but this looks to me like the test isn't really testing an actual updated consent state.

If I'm reading this correctly, the test case starts with `defaultAdPersonalizationConsentSDK` being set to `Denied` via the server configuration, and then `kitInstance.onConsentStateUpdated` us called with `locationCollectionConsent` being set to `false`.

If `kitInstance.onConsentStateUpdated(state, state, filteredMParticleUser)` is never called, will the assertion still be accurate?